### ISSUE TITLE
[java] UnnecessaryFullyQualifiedName: false-positive in an enum that uses its own static variables

### ISSUE THREAD

**Issue by garydgregory** [pmd/pmd#5263]
**Affects PMD Version:** 7.6.0

**Rule:**

[UnnecessaryFullyQualifiedName](https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#unnecessaryfullyqualifiedname)

**Description:**

**Code Sample demonstrating the issue:**

```java
package org.apache.commons.io;

public enum EnumX {

    VALUE_X(EnumX.X); // MUST qualify the name or the file won't compile

    static final String X = "X";

    private final String v;

    EnumX(final String v) {
        this.v = v;
    }

    public String getV() {
        return v;
    }
}
```
Incorrectly generates:
```xml
<violation beginline="5" endline="5" begincolumn="13" endcolumn="18" rule="UnnecessaryFullyQualifiedName" ruleset="Code Style" package="org.apache.commons.io" class="EnumX" externalInfoUrl="https://docs.pmd-code.org/pmd-doc-7.6.0/pmd_rules_java_codestyle.html#unnecessaryfullyqualifiedname" priority="4">
Unnecessary qualifier 'EnumX': 'X' is already in scope because it is declared in an enclosing type
</violation>
```

**Expected outcome:**

PMD reports a violation at line 5, but that's wrong. That's a false positive.

**Running PMD through:** *Apache Maven 3.9.9 (8e8579a9e76f7d015ee5ec7bfcdc97d260186937)*

**Comment by oowekyala** [comment]
Thank you for the report. It seems this is also a problem with regular classes. The problem is the forward reference, which may happen in any type of class/enum/interface/etc.

**Comment by garydgregory** [comment]
Hi @oowekyala 

Its worth noting that in an enum, you don't have the choice but to write the code this way.

I am looking forward to a fix so I can drop all the NOPMD comments I just had to insert 😉 

TY! 😄
